status/condition_set.go

```diff
@@ -237,3 +239,62 @@ func (c ConditionSet) findUnhealthyDependents() []Condition {
 	})
 	return conditions
 }
+
+// unstructuredAdapter is an adapter for the status.Object interface. unstructuredAdapter
+// makes the assumption that status conditions are found on status.conditions path.
+type unstructuredAdapter struct {
+	*unstructured.Unstructured
+}
+
+// FromUnstructured makes the assumption that the status conditions are found on status.conditions
+// path. If they are not found then we return nil.
+func FromUnstructured(u *unstructured.Unstructured) (Object, error) {
+	c, found, err := unstructured.NestedSlice(u.Object, "status", "conditions")
+	if err != nil || !found {
+		return nil, fmt.Errorf("unable to determine status conditions: %w", err)
+	}
+	for _, condition := range c {
+		var newCondition Condition
+		cond := condition.(map[string]interface{})
+		jsonStr, err := json.Marshal(cond)
+		if err != nil {
+			return nil, err
+		}
+		if err = json.Unmarshal(jsonStr, &newCondition); err != nil {
+			return nil, err
+		}
+	}
+	return &unstructuredAdapter{Unstructured: u}, nil
+}
+
+func (u *unstructuredAdapter) GetConditions() []Condition {
+	var conditions []Condition
+	c, _, _ := unstructured.NestedSlice(u.Object, "status", "conditions")
+	for _, condition := range c {
+		var newCondition Condition
+		cond := condition.(map[string]interface{})
+		jsonStr, _ := json.Marshal(cond)
+		json.Unmarshal(jsonStr, &newCondition)
+		conditions = append(conditions, newCondition)
+	}
+	return conditions
+}
+
+func (u *unstructuredAdapter) SetConditions(conditions []Condition) {
+	uc, _, _ := unstructured.NestedSlice(u.Object, "status", "conditions")
+	lo.Map(conditions, func(condition Condition, _ int) []interface{} {
+		var b map[string]interface{}
+		j, _ := json.Marshal(&condition)
+		json.Unmarshal(j, &b)
+		uc = append(uc, b)
+		return uc
+	})
+	unstructured.SetNestedSlice(u.Object, uc, "status", "conditions")
```

Comment:
```suggestion
	unstructured.SetNestedSlice(u.Object, lo.Map(conditions, func(condition Condition, _ int) []interface{} {
		var b map[string]interface{}
		j, _ := json.Marshal(&condition)
		json.Unmarshal(j, &b)
		return b
	}), "status", "conditions")
```

nit: consider the following